File: backend/src/main/java/com/example/backend/entity/Product.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "products")
@Data
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private Double price;

    @Column(name = "image_name")
    private String imageName;

    @Column(name = "category_id")
    private Long categoryId;

    @Column(columnDefinition = "TEXT")
    private String ingredients;

    @Column(name = "created_at", updatable = false)
    @CreationTimestamp
    private LocalDateTime createdAt;
}

File: frontend/src/types/product.ts
export interface Product extends CreateProductPayload {
  id: number
  categoryName: string
  createdAt: string
}

export interface CreateProductPayload {
  name: string
  description: string
  ingredients: string
  price: number
  categoryId?: number
  imageName: string
}

export interface UpdateProductPayload {
  name: string
  description: string
  ingredients: string
  price: number
  categoryId?: number
  imageName: string
}
